### PR TITLE
libelf: enable full elfutils package

### LINKED ISSUE #7409: [Package]:  Elfutils debuginfod
### Package description

elfutils debuginfod is a client/server in elfutils 0.178+ that automatically distributes elf/dwarf/source-code from servers to clients such as debuggers across HTTP

This is already used by projects such as Fedora, debian, Ubuntu and arch Linux will eventually support it. This allows users to debug issues in repo packages without rebuilding it themselves

### Home page URL

https://sourceware.org/elfutils/Debuginfod.html

### Source code URL

https://sourceware.org/elfutils/ftp

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

I have tried to compile this.  _obstack_free is normally part of the gnu libc and I'm not sure how termux handles this

### PR DESCRIPTION
**Pull request by MrAdityaAlok**
- enabled full package
- updated to latest version
- removed few non-required patches
- added hsearch patch and few more patches
- added patch in `argp` to prevent segfault
- defined subpackages for `elfutils`(bin) , `libdebuginfod`, `debuginfod`(bin)
- closes #7409

### CODE REVIEW

**Comment by MrAdityaAlok** [comment]
And I do not know what this warning is:
``
/home/builder/.termux-build/_cache/android-r21d-api-24-v5/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: warning: using 'ELFUTILS_0.122' as version for 'dwarf_bytesize' which is also named in version 'ELFUTILS_0.143' in script
``
So if somebody could help, whether this is harmful for harmless...

File: packages/libelf/debuginfod-client.c.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
+--- elfutils-0.185/debuginfod/debuginfod-client.c	2021-05-22 23:55:24.000000000 +0530
++++ elfutils-0.185-patch/debuginfod/debuginfod-client.c	2021-08-28 22:14:21.559999738 +0530
+@@ -85,6 +85,20 @@
+ #include <sys/utsname.h>
+ #include <curl/curl.h>
+ 
++int futimes(int fd, const struct timeval tv[2]) {
```
Comment (xeffyr) [inline packages/libelf/debuginfod-client.c.patch:7]: Why not to declare it with keyword `static`?

File: packages/libelf/debuginfod.cxx.patch [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ #include <microhttpd.h>
++#include <netinet/in.h>
+ 
+ #if MHD_VERSION >= 0x00097002
+ // libmicrohttpd 0.9.71 broke API
+@@ -116,6 +116,19 @@
+ #define tid() pthread_self()
+ #endif
+ 
++int futimes(int fd, const struct timeval tv[2]) {
```
Comment (xeffyr) [inline packages/libelf/debuginfod.cxx.patch:23]: Why not to declare it with keyword `static`?

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/libelf/debuginfod-client.c.patch (hunk 1 above)
Comment (MrAdityaAlok) [reply packages/libelf/debuginfod-client.c.patch:7]: Oh yes....updating it.

**Review by xeffyr — APPROVED** [review]

**Pull request merged**